From 7b37e4d5f5c81527058635bbeddfac348c981a95 Mon Sep 17 00:00:00 2001
From: Ethan Cotterell <e.cotterell@lancaster.ac.uk>
Date: Mon, 26 Apr 2021 14:07:44 +0100
Subject: [PATCH 1/2] src/fping.c: renamed constants

---
 src/fping.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/src/fping.c b/src/fping.c
index 532fa1c..b1ef3cb 100644
--- a/src/fping.c
+++ b/src/fping.c
@@ -2110,7 +2110,7 @@ int decode_icmp_ipv4(

     icp = (struct icmp*)(reply_buf + hlen);

-    if (icp->icmp_type != ICMP_ECHOREPLY) {
+    if (icp->icmp_type != ICMP_ER) {
         /* Handle other ICMP packets */
         struct icmp* sent_icmp;
         SEQMAP_VALUE* seqmap_value;
@@ -2138,7 +2138,7 @@ int decode_icmp_ipv4(
         getnameinfo(response_addr, response_addr_len, addr_ascii, INET6_ADDRSTRLEN, NULL, 0, NI_NUMERICHOST);

         switch (icp->icmp_type) {
-        case ICMP_UNREACH:
+        case ICMP_DUR:
             h = table[seqmap_value->host_nr];
             if (icp->icmp_code > ICMP_UNREACH_MAXTYPE) {
                 print_warning("ICMP Unreachable (Invalid Code) from %s for ICMP Echo sent to %s",
@@ -2153,10 +2153,10 @@ int decode_icmp_ipv4(
             num_othericmprcvd++;
             break;

-        case ICMP_SOURCEQUENCH:
-        case ICMP_REDIRECT:
-        case ICMP_TIMXCEED:
-        case ICMP_PARAMPROB:
+        case ICMP_SQ:
+        case ICMP_RD:
+        case ICMP_TE:
+        case ICMP_PP:
             h = table[seqmap_value->host_nr];
             if (icp->icmp_type <= ICMP_TYPE_STR_MAX) {
                 print_warning("%s from %s for ICMP Echo sent to %s",
@@ -2230,7 +2230,7 @@ int decode_icmp_ipv6(
         getnameinfo(response_addr, response_addr_len, addr_ascii, INET6_ADDRSTRLEN, NULL, 0, NI_NUMERICHOST);

         switch (icp->icmp6_type) {
-        case ICMP_UNREACH:
+        case ICMP_DUR:
             h = table[seqmap_value->host_nr];
             if (icp->icmp6_code > ICMP_UNREACH_MAXTYPE) {
                 print_warning("ICMP Unreachable (Invalid Code) from %s for ICMP Echo sent to %s",
@@ -2245,10 +2245,10 @@ int decode_icmp_ipv6(
             num_othericmprcvd++;
             break;

-        case ICMP_SOURCEQUENCH:
-        case ICMP_REDIRECT:
-        case ICMP_TIMXCEED:
-        case ICMP_PARAMPROB:
+        case ICMP_SQ:
+        case ICMP_RD:
+        case ICMP_TE:
+        case ICMP_PP:
             h = table[seqmap_value->host_nr];
             if (icp->icmp6_type <= ICMP_TYPE_STR_MAX) {
                 print_warning("%s from %s for ICMP Echo sent to %s",
--
2.25.1


From 349844118639d74a8eabe8c7b44a10d1f355d429 Mon Sep 17 00:00:00 2001
From: Ethan Cotterell <e.cotterell@lancaster.ac.uk>
Date: Mon, 26 Apr 2021 14:20:59 +0100
Subject: [PATCH 2/2] src/fping.c: renamed constants

---
 src/fping.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fping.c b/src/fping.c
index b1ef3cb..ad84ae4 100644
--- a/src/fping.c
+++ b/src/fping.c
@@ -2202,7 +2202,7 @@ int decode_icmp_ipv6(

     icp = (struct icmp6_hdr*)reply_buf;

-    if (icp->icmp6_type != ICMP6_ECHO_REPLY) {
+    if (icp->icmp6_type != ICMP6_TYPE_EREP) {
         /* Handle other ICMP packets */
         struct icmp6_hdr* sent_icmp;
         SEQMAP_VALUE* seqmap_value;
--
2.25.1
