From fb3eb80bd91e2a8107a5d9542c2a6ac1bde6349f Mon Sep 17 00:00:00 2001
From: Ethan Cotterell <e.cotterell@lancaster.ac.uk>
Date: Mon, 26 Apr 2021 15:12:07 +0100
Subject: [PATCH 09/14] src/fping.c: fixed typo in sent_icmp

---
 src/fping.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fping.c b/src/fping.c
index 644a20f..f4c22b6 100644
--- a/src/fping.c
+++ b/src/fping.c
@@ -2087,7 +2087,7 @@ int decode_icmp_ipv4(
             return -1;
         }
 
-        seqmap_value = seqmap_fetch(ntohs(sent_icmp->icmp_seq), current_time_ns);
+        seqmap_value = seqmap_fetch(ntohs(sent_icmp->seqno), current_time_ns);
         if (seqmap_value == NULL) {
             return -1;
         }
-- 
2.25.1

