From 57b3ae21191cd621762e27b8073b4dc08a6a88b4 Mon Sep 17 00:00:00 2001
From: Ethan Cotterell <e.cotterell@lancaster.ac.uk>
Date: Mon, 3 May 2021 16:41:06 +0100
Subject: [PATCH 11/15] src/fping.c: fixed incorrect ip header length

---
 src/fping.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fping.c b/src/fping.c
index f434573..ed962f1 100644
--- a/src/fping.c
+++ b/src/fping.c
@@ -2053,7 +2053,7 @@ int decode_icmp_ipv4(
          */
         hlen = (ip->_v_hl & 0x0F) << 2;
 #else
-        hlen = ip->_v_hl << 2;
+        hlen = (ip->_v_hl & 0x0F) << 2;
 #endif
     }
 
-- 
2.30.2

